staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/validation_test.go

```diff
@@ -424,6 +425,7 @@ func TestValidateCustomResource(t *testing.T) {
 		},
 		{name: "immutability transition rule",
 			schema: apiextensions.JSONSchemaProps{
+				Type: "object",
```

Comment:
curious what changed that made these be required?

Reply:
These tests failed due to https://github.com/kubernetes/kubernetes/blob/d6f4a5d04a93500477dbb1f5df7e9acc42836383/staging/src/k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model/schemas.go#L106 not matching.  I can't remember why exactly it wasn't observed before, but it has something to do with how we build DeclTypes earlier now and traverse through them instead of building them (possibly many times) when needed.